neuro_flow/context.py

```diff
@@ -1023,31 +1057,49 @@ async def _include(
             ret.append({k: await v.eval(self) for k, v in dct.items()})
         return ret
 
+    async def _build_cache(
+        self, ast_cache: Optional[ast.Cache], default: ast.CacheStrategy
+    ) -> CacheCtx:
+        if ast_cache is None:
+            return self.cache
+        strategy = ast_cache.strategy
+        if strategy is None:
+            strategy = default
+        if strategy == ast.CacheStrategy.INHERIT:
+            strategy = self.cache.strategy
+        life_span = await ast_cache.life_span.eval(self)
+        if life_span is None:
+            life_span = self.cache.life_span
+        else:
+            life_span = min(self.cache.life_span, life_span)
+        return CacheCtx(strategy=strategy, life_span=life_span)
+
 
 @dataclass(frozen=True)
 class BatchContext(TaskContext, BaseFlowContext):
     FLOW_TYPE: ClassVar[Type[ast.BatchFlow]] = field(init=False, default=ast.BatchFlow)
     LOOKUP_KEYS: ClassVar[Tuple[str, ...]] = field(
         init=False,
-        default=BaseFlowContext.LOOKUP_KEYS
-        + TaskContext.LOOKUP_KEYS
-        + ("args", "needs"),
+        default=BaseFlowContext.LOOKUP_KEYS + TaskContext.LOOKUP_KEYS + ("args",),
     )
     _args: Optional[Mapping[str, str]] = None
 
     @classmethod
     async def create(
         cls: Type[_CtxT],
         ast_flow: ast.BaseFlow,
+        flow_digest: Digest,
         workspace: LocalPath,
         config_file: LocalPath,
     ) -> _CtxT:
         ctx = await super(cls, BatchContext).create(
             ast_flow,
+            flow_digest,
             workspace,
             config_file,
         )
         ast_defaults = ast_flow.defaults
+        ctx = replace(ctx, _cache=CacheCtx())
```

Comment:
Why do we need defaults cache context to be available while we process defaults? It will allow very strange code:
```
defaults:
  cache:
    strategy: ${{cache.strategy}}
```

Can we set the default in the else branch below (line 1117)?

Reply:
Oh, I see, the `self._build_cache` uses `self.cache`. Do not see good clean solution here, but we can at least move `replace` call below `eval` calls.

Reply:
I was thinking about pushing the project default cache strategy in `project.yml` but did not implement it yet.
The example makes value with the mentioned case.